test/c3/appframework/appframework_test.go

```diff
@@ -211,4 +212,83 @@ var _ = Describe("c3appfw test", func() {
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListV1, false, true)
 		})
 	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("c3, appframework: can deploy a C3 SVA and have ES app installed on SHC", func() {
+
+			// ES is a huge file, we configure it here rather than in BeforeSuite/BeforeEach to save time for other tests
+			// Upload ES app to S3
+			appListV1 = testenv.ESApp
+			appFileList := testenv.GetAppFileList(appListV1, 1)
+
+			// Download V1 Apps from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download V1 app files")
+
+			// Upload V1 apps to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload ES app to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec
+			// volumeSpec: Volume name, Endpoint, Path and SecretRef
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterprisev1.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "cluster",
+			}
+
+			// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+
+			// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create Single site Cluster and SHC, with App Framework enabled on CM and SHC Deployer
+			shSpec := enterprisev1.SearchHeadClusterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					ExtraEnv: []corev1.EnvVar{
+						{
+							Name:  "SPLUNK_ES_SSL_ENABLEMENT",
+							Value: "ignore"},
+					},
+					Volumes:                      []corev1.Volume{},
+					LivenessInitialDelaySeconds:  1450,
```

Comment:
How did we come to this value 1450? :)

Reply:
through successive runs, seeing the "Liveness probe failed" message with lower values.
To give an idea of the duration of each step, here's the logs on the deployer on latest successful run:
```
Monday 12 July 2021  19:05:05 +0000 (0:00:01.571)       0:21:13.423 ***********
===============================================================================
splunk_deployer : Apply shcluster bundle ------------------------------ 825.65s
splunk_deployer : Wait for SHC to be ready ---------------------------- 130.23s
splunk_deployer : Run ESS post-install setup -------------------------- 100.13s
splunk_common : Restart the splunkd service - Via CLI ------------------ 29.66s
splunk_common : Restart the splunkd service - Via CLI ------------------ 23.43s
splunk_common : Start Splunk via CLI ----------------------------------- 23.37s
splunk_common : Restart the splunkd service - Via CLI ------------------ 22.80s
splunk_deployer : Disable installed apps ------------------------------- 13.92s
splunk_deployer : Install app via REST --------------------------------- 11.79s
splunk_deployer : Copy installed apps to /opt/splunk/etc/shcluster/apps --- 8.09s
splunk_deployer : Install app via REST ---------------------------------- 7.54s
splunk_deployer : Check app contents ------------------------------------ 6.51s
splunk_deployer : Install app via REST ---------------------------------- 4.42s
splunk_deployer : Check local app --------------------------------------- 4.17s
splunk_deployer : Install app via REST ---------------------------------- 4.14s
splunk_deployer : Install app via REST ---------------------------------- 3.18s
splunk_common : Get Splunk status --------------------------------------- 2.51s
splunk_common : Update /opt/splunk/etc ---------------------------------- 1.98s
splunk_common : Test basic https endpoint ------------------------------- 1.91s
splunk_deployer : Check for required restarts --------------------------- 1.66s
===============================================================================
```

Reply:
This seems like an environment specific issue. It might need tuning when test runs on nightly build. Please keep an eye out